x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentRebalancer.java

```diff
@@ -81,7 +86,56 @@ TrainedModelAssignmentMetadata.Builder rebalance() {
         }
 
         AssignmentPlan assignmentPlan = computeAssignmentPlan();
-        return buildAssignmentsFromPlan(assignmentPlan);
+        TrainedModelAssignmentMetadata.Builder builder = buildAssignmentsFromPlan(assignmentPlan);
+
+        if (shuttingDownNodeIds.isEmpty()) {
+            return builder;
+        }
+
+        return setShuttingDownNodeRoutesToStopping(builder);
```

Comment:
This shouldn't be necessary. Nodes that are shutting down are not included in the `Map<DiscoveryNode, NodeLoad> nodeLoads` parameter, the planner should never assigned to nodes that are shutting down.